File: app/src/main/java/com/tejasbhitle/gallery/fragment/SettingsFragment.java
package com.tejasbhitle.gallery.fragment;

import android.os.Bundle;
import android.util.Log;

import com.tejasbhitle.gallery.R;

import androidx.preference.ListPreference;
import androidx.preference.Preference;
import androidx.preference.PreferenceFragmentCompat;
import androidx.preference.PreferenceManager;

public class SettingsFragment extends PreferenceFragmentCompat {

    private static final String TAG = "SettingsFragment";

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {

        addPreferencesFromResource(R.xml.settings_prefs);

        bindPreferenceSummaryToValue(findPreference(getString(R.string.prefs_theme)));
    }

    private void bindPreferenceSummaryToValue(Preference preference) {
        preference.setOnPreferenceChangeListener(bindPreferenceSummaryToValueListener);

        bindPreferenceSummaryToValueListener.onPreferenceChange(preference,
                PreferenceManager
                        .getDefaultSharedPreferences(preference.getContext())
                        .getString(preference.getKey(), ""));
    }

    private Preference.OnPreferenceChangeListener
            bindPreferenceSummaryToValueListener = new Preference.OnPreferenceChangeListener() {
        @Override
        public boolean onPreferenceChange(Preference preference, Object newValue) {

            String stringValue = newValue.toString();

            if(preference instanceof ListPreference){
                ListPreference listPreference = (ListPreference)preference;
                int index = listPreference.findIndexOfValue(stringValue);

                preference.setSummary(
                        index >= 0
                        ? listPreference.getEntries()[index]
                        : null);
            }
            return true;
        }
    };
}

File: app/src/main/java/com/tejasbhitle/gallery/fragment/SettingsFragment.kt
package com.tejasbhitle.gallery.fragment

import android.os.Bundle
import android.util.Log
import androidx.fragment.app.DialogFragment
import androidx.preference.*
import com.tejasbhitle.gallery.R
import com.tejasbhitle.gallery.preference.ColumnDialogPreference
import com.tejasbhitle.gallery.preference.ColumnPreferenceDialogFragmentCompat

class SettingsFragment : PreferenceFragmentCompat() {

    private val TAG = "SettingsFragment"

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {

        addPreferencesFromResource(R.xml.settings_prefs)
        bindPreferenceSummaryToValue(findPreference(getString(R.string.prefs_theme)))
    }

    private fun bindPreferenceSummaryToValue(preference: Preference) {
        preference.onPreferenceChangeListener = bindPreferenceSummaryToValueListener

        bindPreferenceSummaryToValueListener.onPreferenceChange(preference,
                PreferenceManager
                        .getDefaultSharedPreferences(preference.context)
                        .getString(preference.key, ""))
    }

    private val bindPreferenceSummaryToValueListener = Preference.OnPreferenceChangeListener {
        preference, newValue ->
        val stringValue = newValue.toString()

        if (preference is ListPreference) {
            val index = preference.findIndexOfValue(stringValue)

            preference.setSummary(
                    if (index >= 0) preference.entries[index]
                    else null
            )
        }
        true
    }

    override fun onDisplayPreferenceDialog(preference: Preference) {

        Log.e(TAG,"preference not null")

        var dialogFragment: DialogFragment? = null
        if(preference is ColumnDialogPreference){
            dialogFragment = ColumnPreferenceDialogFragmentCompat
                    .newInstance(preference)
        }

        if(dialogFragment != null){
            dialogFragment.setTargetFragment(this,0)
            dialogFragment.show(this.fragmentManager,"PREFERENCE_DIALOG")
        }
        else
            super.onDisplayPreferenceDialog(preference)
    }
}